src/components/RoomHeaderAvatars.js

```diff
@@ -6,46 +6,68 @@ import styles from '../styles/styles';
 import Text from './Text';
 import CONST from '../CONST';
 import Avatar from './Avatar';
+import themeColors from '../styles/themes/default';
 
 const propTypes = {
-    /** Array of avatar URL */
-    avatarImageURLs: PropTypes.arrayOf(PropTypes.string),
+    /** Array of avatar URLs or icons */
+    avatarIcons: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.func])),
 
-    /** Whether this avatar is for a custom/default room */
-    isChatRoom: PropTypes.bool,
+    /** Whether show large Avatars */
+    shouldShowLargeAvatars: PropTypes.bool,
 };
 
 const defaultProps = {
-    avatarImageURLs: [],
-    isChatRoom: false,
+    avatarIcons: [],
+    shouldShowLargeAvatars: false,
 };
 
 const RoomHeaderAvatars = (props) => {
-    if (!props.avatarImageURLs.length) {
+    if (!props.avatarIcons.length) {
         return null;
     }
 
-    if (props.avatarImageURLs.length === 1 || props.isChatRoom) {
+    if (props.avatarIcons.length === 1) {
         return (
             <Avatar
-                source={props.avatarImageURLs[0]}
+                source={props.avatarIcons[0]}
                 imageStyles={[styles.avatarLarge]}
-                isChatRoom={props.isChatRoom}
+                fill={themeColors.iconSuccessFill}
+                size={CONST.AVATAR_SIZE.LARGE}
             />
         );
     }
 
-    // avatarImageURLsToDisplay
-    const avatarImageURLsToDisplay = props.avatarImageURLs.slice(0, CONST.REPORT.MAX_PREVIEW_AVATARS);
+    if (props.shouldShowLargeAvatars) {
+        return (
+            <View style={[styles.flexRow, styles.wAuto, styles.justifyContentCenter, styles.alignItemsCenter]}>
+                <View style={styles.leftSideLargeAvatar}>
+                    <Avatar
+                        source={props.avatarIcons[1]}
+                        imageStyles={[styles.avatarLarge]}
+                        size={CONST.AVATAR_SIZE.LARGE}
+                        fill={themeColors.iconSuccessFill}
+                    />
+                </View>
+                <View style={[styles.secondAvatarHovered, styles.rightSideLargeAvatar]}>
```

Comment:
I've looked into it, and it seems like `secondAvatarHovered` is a poor name for this style:

```js
secondAvatarHovered: {
    backgroundColor: themeColors.sidebarHover,
    borderColor: themeColors.sidebarHover,
}
```

It's used in several locations that don't have much to do with hovering. What this style does is provide the white background around the secondary avatars to create a visual space between avatars.

<img width="118" alt="image" src="https://user-images.githubusercontent.com/47436092/156233705-12d50d00-e996-460d-9aee-7405181e1490.png">

So I would rename it to something else that's more accurately representative of the style. One option would be to just rename it to `secondAvatarBackground`, but another option would be to rename `defaultModalContainer` to just `defaultComponentBG` and reuse that in place of `secondAvatarHovered`.

Reply:
Another option would be to use `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`

Reply:
Oh makes sense, I'm going to replace `secondAvatarHovered` with `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`